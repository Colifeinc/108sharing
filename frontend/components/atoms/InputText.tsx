import * as React from 'react'
import FormItem from 'components/layouts/FormItem'
import classNames from 'classnames'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { ReactNode } from 'react'
interface IProps {
  name: string
  defaultValue: string
  readonly?: boolean
  label?: null | string
  placeholder?: null | string
  error?: null | string
  required?: boolean
  type?: string
  inputRef?: any
  onChangeHandler?(event: any): void | Promise<void>
  onBlurHandler?(event: any): void | Promise<void>
  icon?: ReactNode
}

const InputText: React.FC<IProps> = ({
  name,
  defaultValue,
  readonly,
  label,
  placeholder,
  error,
  required,
  type,
  inputRef,
  onChangeHandler,
  onBlurHandler,
  icon,
}) => {
  return (
    <FormItem label={label} required={required} error={error}>
      <div className="relative w-ful">
        {icon && (
          <div className="absolute left-0 inset-y-0 flex items-center ml-3 text-neutral-600 pointer-events-none">
            {icon}
          </div>
        )}
        <input
          className={classNames([
            'block w-full px-3 py-2 bg-neutral-100 rounded outline-none appearance-none',
            'border-2 border-transparent',
            !readonly && 'focus:border-2 focus:border-primary focus:bg-white',
            readonly && 'text-neutral-500 cursor-default',
            icon && 'pl-8',
          ])}
          id={name}
          type={type ?? 'text'}
          name={name}
          placeholder={placeholder}
          required={required ? true : false}
          defaultValue={defaultValue}
          onChange={event => {
            if (typeof onChangeHandler === 'function') {
              onChangeHandler(event)
            }
          }}
          onBlur={event => {
            if (typeof onBlurHandler === 'function') {
              onBlurHandler(event)
            }
          }}
          ref={inputRef || null}
          readOnly={readonly || false}
        />
      </div>
    </FormItem>
  )
}

export default InputText
