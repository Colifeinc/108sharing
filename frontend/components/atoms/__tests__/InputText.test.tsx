// tslint:disable-next-line
import '../../../test/matchMedia.mock'

import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import * as React from 'react'
import InputText from '../InputText'

jest.mock('../../../core/i18n')

describe('atoms/InputText.tsx', () => {
  const onChangeHandler = jest.fn()
  const onBlurHandler = jest.fn()
  const props = {
    name: 'test',
    defaultValue: 'test',
    onChangeHandler,
    onBlurHandler,
  }

  it('should matches to the snapshot', () => {
    const wrapper = shallow(<InputText {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  // it('should matches to the snapshot with label', () => {
  //   const wrapper = shallow(<InputText {...props} label="Label" />)
  //   expect(wrapper.find('.InputText_Title').length).toBe(1)
  //   expect(toJson(wrapper)).toMatchSnapshot()
  // })

  // it('should matches to the snapshot with required', () => {
  //   const wrapper = shallow(<InputText {...props} label="Label" required={true} />)
  //   expect(wrapper.find('.InputText_Required').length).toBe(1)
  //   expect(toJson(wrapper)).toMatchSnapshot()
  // })

  // it('should matches to the snapshot with error', () => {
  //   const wrapper = shallow(<InputText {...props} error={'error'} />)
  //   expect(wrapper.find('.InputText_Error').length).toBe(1)
  //   expect(toJson(wrapper)).toMatchSnapshot()
  // })

  // it('should call handlers', () => {
  //   const wrapper = shallow(<InputText {...props} />)
  //   wrapper.find('input[type="text"]').simulate('change')
  //   expect(onChangeHandler).toHaveBeenCalled()
  //   wrapper.find('input[type="text"]').simulate('blur')
  //   expect(onBlurHandler).toHaveBeenCalled()
  // })
})
