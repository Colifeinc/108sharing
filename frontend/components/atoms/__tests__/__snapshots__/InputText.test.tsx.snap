// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/InputText.tsx should matches to the snapshot 1`] = `
<FormItem>
  <div
    className="relative w-ful"
  >
    <input
      className="block w-full px-3 py-2 bg-neutral-100 rounded outline-none appearance-none border-2 border-transparent focus:border-2 focus:border-primary focus:bg-white"
      defaultValue="test"
      id="test"
      name="test"
      onBlur={[Function]}
      onChange={[Function]}
      readOnly={false}
      required={false}
      type="text"
    />
  </div>
</FormItem>
`;
